FamilySearchIosSampleApp/TreeTVC.swift

```diff
@@ -0,0 +1,249 @@
+//
+//  TreeVC.swift
+//  FamilySearchIosSampleApp
+//
+//  Created by Eduardo Flores on 6/6/16.
+//  Copyright © 2016 FamilySearch. All rights reserved.
+//
+
+import UIKit
+
+class TreeTVC: UITableViewController {
+    
+    @IBOutlet weak var activityIndicator: UIActivityIndicatorView!
+    
+    var user : User!
+    
+    var personArray = NSArray()
+    
+    var accessToken : String!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // get the access token from NSUserDefaults
+        let preferences = NSUserDefaults.standardUserDefaults()
+        accessToken = preferences.stringForKey(Utilities.KEY_ACCESS_TOKEN)
+        
+        // get url for family tree from Collections
+        Utilities.getUrlsFromCollections({ (collectionsResponse, error) -> Void in
```

Comment:
Rather than fight this async loading of your urls problem every time you want to make a request, it seems like it would work better to just load these urls before they ever get to the login form (or maybe when the login form is displayed) and then store them somewhere they can be directly referenced as non-optional values.  The app won't work if these urls don't load, so it seems like a reasonable thing require finishes loading before the UI allows the user to do anything.  The response should be cached I would think so the user doesn't take that hit every time they open the app but for that initial load, I'd kick it off right when the login form shows and then don't allow them to login until the urls have been loaded (since they can't login until after that anyway)